### PR TITLE
Support for WebTestClient

### LINKED ISSUE #238: Support for WebTestClient
As Mentioned in #176 the new WebTestClient should be supported. Otherwise Spring Auto REST Docs can't be used in "new" Spring applications which use the WebTestClient instead of MockMvc to test reactive WebFlux endpoints.

Currently there is some "magic" in `JacksonResultHandlers.JacksonPreparingResultHandler#handle(MvcResult)`. This method only works in a mvc environment. I think an alternative for this is the first step to support WebTestClient (PR is on the way).

### PR DESCRIPTION
**Pull request by Josef-Reichardt**
Closes #238.

### CODE REVIEW

File: spring-auto-restdocs-core/pom.xml [hunk 1]
```diff
@@ -89,7 +94,6 @@
             <groupId>org.apache.tomcat.embed</groupId>
             <artifactId>tomcat-embed-el</artifactId>
             <version>9.0.2</version>
-            <scope>test</scope>
```
Comment (fbenz) [inline spring-auto-restdocs-core/pom.xml]: Is there a reason why this was changed? I think that the test scope is sufficient. We want to force as few dependencies on Spring Auto REST Docs clients as possible.

Comment (Josef-Reichardt) [reply spring-auto-restdocs-core/pom.xml]: Absolutely. Can't remember why I have removed the `test` scope. I think I have tried something... I have added it scope again now.

**Comment by Josef-Reichardt** [comment]
Neither Mono nor Flux was supported correctly. I have adjusted the example application now to have one method which uses `Mono<...>` and one which uses `Flux<...>` as return type and realized that for both methods the `auto-response-fields` snippet was empty. To resolve this issue I have added the two types in `JacksonResponseFieldSnippet`. Now it looks good for Mono **and** Flux.

File: spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/payload/JacksonResponseFieldSnippet.java [hunk 2]
```diff
@@ -71,6 +71,19 @@ protected Type getType(final HandlerMethod method) {
         } else if ("void".equals(returnType.getName())) {
             return null;
         } else {
+            try {
+                Class<?> monoClass = Class.forName("reactor.core.publisher.Mono");
+                Class<?> fluxClass = Class.forName("reactor.core.publisher.Flux");
```
[the remark this answers is not in the document]
Comment (Josef-Reichardt) [reply spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/payload/JacksonResponseFieldSnippet.java]: Yes it's the same here. And yes it's really better/cleaner to do the type check this way and without the try/catch.
I've changed it and also added unit tests for the Mono and Flux return types.

File: samples/java-webtestclient/pom.xml [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-security</artifactId>
+        </dependency>
+        <!--<dependency>-->
+            <!--<groupId>org.springframework.security.oauth</groupId>-->
+            <!--<artifactId>spring-security-oauth2</artifactId>-->
+            <!--<version>2.3.3.RELEASE</version>-->
+        <!--</dependency>-->
```
[the remark this answers is not in the document]
Comment (Josef-Reichardt) [reply samples/java-webtestclient/pom.xml]: Done :+1:

File: samples/java-webtestclient/generated-docs/index.html [hunk 4]
```diff
@@ -0,0 +1,2471 @@
...
+</div>
+<div class="paragraph">
+<p>Updates existing item.</p>
+</div>
+<div class="paragraph">
+<p>An example of returning an {@link HttpEntity}.</p>
+</div>
+<div class="sect3">
+<h4 id="_authorization_4"><a class="anchor" href="#_authorization_4"></a><a class="link" href="#_authorization_4">2.4.1. Authorization</a></h4>
+<div
```
Comment (Josef-Reichardt) [inline samples/java-webtestclient/generated-docs/index.html]: URL is missing in docs if URI is set by UriBuilder.

Example:
```
webTestClient.get()
    .uri(uriBuilder -> uriBuilder.path("/items/search")
        .queryParam("desc", "main")
        .queryParam("hint", "1")
        .build())
    // ...
```
I will look at this.

Comment (Josef-Reichardt) [reply samples/java-webtestclient/generated-docs/index.html]: Fixed: 9bf28b0

**Comment by jmisur** [comment]
let me merge this already. if you find some improvements or bugfixes, just submit a PR. We should be much faster with additional smaller PRs then.

**Pull request merged**